clang_delta/EmptyStructToInt.cpp

```diff
@@ -343,17 +437,120 @@ bool EmptyStructToInt::isValidRecordDecl(const RecordDecl *RD)
 
   const DeclContext *Ctx = dyn_cast<DeclContext>(CXXDef);
   TransAssert(Ctx && "Invalid DeclContext!");
+  int count = 0;
   for (DeclContext::decl_iterator I = Ctx->decls_begin(),
        E = Ctx->decls_end(); I != E; ++I) {
+    if ((*I)->isReferenced())
+        return false;
```

Comment:
I tried your example and it is correctly transformed to `int s = 0;`. I tested with file extension .cc, .cpp and .c.
But I think you are right that it is better to check only for the explicit member whether they are referenced.